pkg/operator/ceph/cluster/watcher.go

```diff
@@ -511,8 +514,8 @@ func concatenateWatcherIp(address string) string {
 	return watcherIP
 }
 
-func fenceResourceName(nodeName, driver string) string {
-	return fmt.Sprintf("%s-%s", nodeName, driver)
+func fenceResourceName(nodeName, driver, namespace string) string {
+	return fmt.Sprintf("%s-%s-%s", nodeName, driver, namespace)
```

Comment:
If we're changing the resource name, are we going to have upgraded clusters that have duplicate network fence resources for both the old and new names?

Reply:
@Madhu-1 @subhamkrai Will this be an issue with upgraded clusters?

Reply:
Yes, Madhu is looking on how to overcome this

Reply:
Me and @Madhu-1 discussed in call offline, and we concluded that we don't need think about older networkFence since it has the `ownerReference` so Kubernetes will delete that in most probably and let's say if some users still face that we'll suggest them to manually delete that. As it is very rare and there are high changes kubernetes garbage collector will remove that even if we have `ownerRef` in clusterResources.

Reply:
If the garbage collector would remove it with the bad ownerRef, I would have expected them to either be deleted immediately, or to never be deleted. Since we didn't see them disappear immediately, likely they will just be left behind and they can be manually deleted.